dispatch/static/manager/src/js/components/PollEditor/Poll.js

```diff
@@ -0,0 +1,128 @@
+import React, { Component } from 'react'
+import { Button, Intent } from '@blueprintjs/core'
+
+require('../../../styles/components/poll.scss')
+
+const COLOR_OPACITY = .8
+
+class Poll extends Component {
+  constructor(props){
+    super(props)
+    this.state = {
+      showResults: true,
+    }
+  }
+
+  componentDidMount() {
+    this.update()
+  }
+
+  update() {
+    let answers = []
+    let votes = []
+    let pollQuestion = this.props.question ? this.props.question : 'Poll Question'
+    if(this.props.answers){
+      for(let answer of this.props.answers){
+        answers.push(answer['name'])
+        votes.push(answer['vote_count'])
+      }
+    } else {
+      answers.push('First answer')
+      answers.push('Second answer')
```

Comment:
These can be combined:

```js
answers.push('First answer', 'Second answer')

Reply:
However, I would move this logic around a bit so that you don't have to check again below. You can assume that if `this.props.answers` is empty then there are no votes yet:

```js
answers.push('First answer', 'Second answer')
votes.push(2, 1)